ReactiveCocoa/Swift/Property.swift

```diff
@@ -89,35 +105,71 @@ public final class MutableProperty<Value>: MutablePropertyType {
 	/// created from the `values` producer.
 	public var value: Value {
 		get {
-			lock.lock()
-			let value = _value
-			lock.unlock()
-			return value
+			return withValue { $0 }
 		}
 
 		set {
-			lock.lock()
-			_value = newValue
-			observer.sendNext(newValue)
-			lock.unlock()
+			modify { _ in newValue }
 		}
 	}
 
+	/// A signal that will send the property's changes over time,
+	/// then complete when the property has deinitialized.
+	public let signal: Signal<Value, NoError>
+
 	/// A producer for Signals that will send the property's current value,
 	/// followed by all changes over time, then complete when the property has
 	/// deinitialized.
-	public let producer: SignalProducer<Value, NoError>
+	public var producer: SignalProducer<Value, NoError> {
```

Comment:
Yeah, you can extract a `skip(1)` signal before sending the initial value in the initialiser to achieve the same goal.

Anyway, the implementation here is thread-safe if I didn't miss something - the start handler has to acquire the recursive lock to do its work (replay the current value + setup the forwarding to the started signal). It should be slightly simpler too, since now you get one less lock and copy of value (BufferState) in the memory.